Frameworks/CoreText/CTFramesetter.mm

```diff
@@ -36,49 +36,71 @@ static id _createFrame(_CTFrameSetter* frameSetter, CGRect frameSize, CGSize* si
         ret->_lines.attach([NSMutableArray new]);
     }
 
-    sizeOut->width = 0;
-    sizeOut->height = 0;
-
-    float y = frameSize.size.height; //[font ascender];
-    CFIndex curIdx = 0;
-    NSString* string = [frameSetter->_typesetter->_attributedString string];
-    CFIndex stringRange = [string length];
-
-    for (;;) {
-        CFIndex pos = CTTypesetterSuggestLineBreakWithOffset((CTTypesetterRef)(_CTTypesetter*)frameSetter->_typesetter,
-                                                             curIdx,
-                                                             frameSize.size.width,
-                                                             0.0f);
-        if (pos == curIdx)
-            break;
-
-        CFRange lineRange;
-        lineRange.location = curIdx;
-        lineRange.length = pos - curIdx;
-
-        CTLineRef line = CTTypesetterCreateLineWithOffset((CTTypesetterRef)(_CTTypesetter*)frameSetter->_typesetter, lineRange, 0.0f);
-
-        float ascent = 0.0f, descent = 0.0f, leading = 0.0f;
-        float width = CTLineGetTypographicBounds(line, &ascent, &descent, &leading);
-        float lineHeight = ascent - descent + leading;
-
-        if (ret) {
-            CGPoint lineOrigin;
-            lineOrigin.x = 0.0f;
-            lineOrigin.y = y - ascent;
-            [ret->_lines addObject:(id)line];
-            ret->_lineOrigins.push_back(lineOrigin);
+    // Only fill in frame if there is text
+    if (CFAttributedStringGetLength(reinterpret_cast<CFAttributedStringRef>(frameSetter->_typesetter->_attributedString))) {
```

Comment:
When I say 'stronger', I mean that reinterpret_cast is more permissive (thus more dangerous) than static_cast. We should use the most restrictive cast possible.

---
In reply to: [74634231](https://github.com/Microsoft/WinObjC/pull/873#discussion_r74634231) [](ancestors = 74634231)